Updating the renderer to 1.21.10

temotskipa opened the pull request:
This PR updates the mod and refactors the renderer for 1.21.10 compatibility and prepares it for future enhancements like RT. It also includes changes to the README, refactors the mod to use Mojang mappings (due to better rendering class mappings) and adds other stuff.

temotskipa commented:
It is still WIP, so I published it as a draft PR for now. If anyone has any suggestions or better yet, code, feel free to contribute.

NXTler commented:
Why do you update Java to 25 and lwjgl to 3.4.0, when minecraft only supports Java 21 and lwjgl 3.3.3? Especially the lwjgl change will cause minecraft to not start at all, as version 3.3.3 is a hard dependency.

NXTler commented:
Focus only on what the PR goal is and don't make unnecessary/broken changes like these. You can make a separate PR for these.

NXTler commented on README.md:
These changes are also worthless and beside the goal. The Readme would even look the same as before.

temotskipa replied:
This is just a consequence of the IDE. IntelliJ reformatted the code like that.

NXTler commented on src/main/java/net/vulkanmod/config/gui/OptionBlock.java:
```diff
@@ -4,4 +4,4 @@
 public record OptionBlock(String title, Option<?>[] options) {
 
-}
+}
```
?

NXTler commented:
You also modify the config and options for no reason. Again, make a separate PR for these changes.

NXTler commented:
You do a bunch of useless formatting changes in the shaders. Put these in a separate PR and stick to the 'old' formatting for now.

NXTler commented on src/main/java/net/vulkanmod/render/texture/SpriteUpdateUtil.java:
```diff
@@ -9,8 +9,8 @@
 
-    private static boolean doUpload = true;
     private static final Set<VulkanImage> transitionedLayouts = new HashSet<>();
+    private static boolean doUpload = true;
```
I'm talking about stuff like this.

temotskipa commented:
I aligned the LWJGL version with MC to prevent classpath conflicts as @NXTler suggested. Let's see how it works out.

temotskipa commented:
> Why do you update Java to 25 and lwjgl to 3.4.0, when minecraft only supports Java 21 and lwjgl 3.3.3? Especially the lwjgl change will cause minecraft to not start at all, as version 3.3.3 is a hard dependency.

I need Java 25 to minimize unsafe memory access, which could cause silent crashes and is also deprecated. Also, using the most recent Java LTS version is standard practice. I also attempted to use a newer LWJGL version because of this, but it seems like that's not possible.

NXTler commented:
Forcing Java 25 on users to play a Java 21 game isn't the solution and unsafe memory accesses shouldn't happen in the first place. This is a completely unnecessary change, as you can see in the recent test builds.

NXTler commented:
You also started adding new features again, put this in different PRs.

temotskipa commented:
> Forcing Java 25 on users to play a Java 21 game isn't the solution and unsafe memory accesses shouldn't happen in the first place. This is a completely unnecessary change, as you can see in the recent test builds.

LWJGL uses sun.misc.Unsafe internally in its MemoryStack and MemoryUtil classes, and VUtil also used it until I changed it to use FFM, so it did happen and I fixed it (well, as much as I could without entirely rewriting LWJGL's memory access methods). And the specific feature in question is necessary to complete the renderer refactor and make it work on 1.21.10.

a1game commented:
> You also started adding new features again, put this in different PRs.

as far as i know about github , i dont think you can really make 2 pr's from one account due to the limit of 1 fork

NXTler commented:
You can have different branches in a fork for different PRs with no issue. He should only focus on doing one thing at a time or his PR will never get merged.

a1game commented:
> You can have different branches in a fork for different PRs with no issue. He should only focus on doing one thing at a time or his PR will never get merged.

its choice of whoever has ability / perm to merge

NXTler commented:
> LWJGL uses sun.misc.Unsafe internally in its MemoryStack and MemoryUtil classes, and VUtil also used it until I changed it to use FFM, so it did happen and I fixed it (well, as much as I could without entirely rewriting LWJGL's memory access methods).

Do you have prove that it ever caused any issues, or did the AI tell you that? The official test builds for VulkanMod 1.21.10 work completely fine with java 21, so what's the point in fixing a not existing issue? Again, this is just forcing the user to use another Java version than the game they are playing.

a1game commented:
> > LWJGL uses sun.misc.Unsafe internally in its MemoryStack and MemoryUtil classes, and VUtil also used it until I changed it to use FFM, so it did happen and I fixed it (well, as much as I could without entirely rewriting LWJGL's memory access methods).
> 
> Do you have prove that it ever caused any issues, or did the AI tell you that? The official test builds for VulkanMod 1.21.10 work completely fine with java 21, so what's the point in fixing a not existing issue? Again, this is just forcing the user to use another Java version than the game they are playing.

"The official test builds for VulkanMod 1.21.10 work completely fine with java 21" are you speaking of test build of this fork but with properties changed to use java 21 instead of this persons java 25 decision?

NXTler commented:
Not at all, Collateral has been working on 1.21.10 for some time now, and we have test builds on the Discord. They have nothing to do with this fork in any way and are functional without Java 25.

temotskipa commented:
> Not at all, Collateral has been working on 1.21.10 for some time now, and we have test builds on the Discord. They have nothing to do with this fork in any way and are functional without Java 25.

Can't Collateral commit those changes to the dev branch if they're mostly functional? I could build on top of his work instead of diverging in ways not intended for the project.

a1game commented:
> Not at all, Collateral has been working on 1.21.10 for some time now, and we have test builds on the Discord. They have nothing to do with this fork in any way and are functional without Java 25.

i dont spend time on that platform therefore i didnt know, i may join.
